lib/os/ring_buffer.c

```diff
@@ -284,3 +284,28 @@ uint32_t ring_buf_get(struct ring_buf *buf, uint8_t *data, uint32_t size)
 
 	return total_size;
 }
+
+uint32_t ring_buf_peek(struct ring_buf *buf, uint8_t *data, uint32_t size)
+{
+	uint8_t *src;
+	uint32_t partial_size;
+	uint32_t total_size = 0U;
+	int err;
+
+	size = MIN(size, ring_buf_size_get(buf));
+
+	do {
+		partial_size = ring_buf_get_claim(buf, &src, size);
```

Comment:
Seems like peeking beyond buffer boundary is not possible. `partial_size` will return 0 if boundary is reached, that will terminate loop and since `ring_buf_get_finish` is called later another peek will not look further but start from the same spot.

It would be good to do one more run when `partial_size` returned 0.

Reply:
I'll double check that there is test coverage for wrapping around the buffer boundary, but that function does work for that purpose for me.

Side note, I am intentionally not asserting things that should be covered elsewhere in the test suite, to avoid duplication.

E.g. the same code has been working for some time in ring_buf_size_get()

Reply:
No problems with peek wrapping but I added test coverage to be safe. It is mainly covered by claim() / finish() with the tmp_head concept. Assuming there is existing test coverage for claim() / finish(), then the additional checks are not really needed.